controller/search.go

```diff
@@ -202,17 +288,20 @@ func (c *SearchController) Users(ctx *app.UsersSearchContext) error {
 
 // Iterate over the WI list and read parent IDs
 // Fetch and load Parent WI in the included list
-func (c *SearchController) enrichWorkItemList(ctx *app.ShowSearchContext, ancestors link.AncestorList, res *app.SearchWorkItemList) {
-	fetchInBatch := []uuid.UUID{}
+func (c *SearchController) enrichWorkItemList(ctx *app.ShowSearchContext, ancestors link.AncestorList, matchingIDs id.Slice, res *app.SearchWorkItemList) {
+
+	parentIDs := id.Slice{}
 	for _, wi := range res.Data {
 		if wi.Relationships != nil && wi.Relationships.Parent != nil && wi.Relationships.Parent.Data != nil {
-			parentID := wi.Relationships.Parent.Data.ID
-			fetchInBatch = append(fetchInBatch, parentID)
+			parentIDs = append(parentIDs, wi.Relationships.Parent.Data.ID)
 		}
 	}
 
-	// Also append the ancestors not already included in the parent list.
-	fetchInBatch = append(fetchInBatch, link.IDSliceDiff(ancestors.GetDistinctAncestorIDs(), fetchInBatch)...)
+	// Also append the ancestors not already included in the parent list
+	parentAncestorDiff := ancestors.GetDistinctAncestorIDs().Diff(parentIDs)
+	fetchInBatch := append(parentIDs, parentAncestorDiff...)
```

Comment:
what about an `Add` function on `id.Slice{}`, since you have a `Sub` ? then you could write something like: 
````
fetchInBatch := parentIDs.Add(parentAncestorDiff...).Sub(matchingIDs)
````
wdyt ?

Reply:
I've added (and tested it in) 09b51912.

Reply:
ok, thanks